src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,583 @@
+        /// Find where we need to insert additional newlines
+        /// </summary>
+        private void FindLineWraps(int cursorPos, List<string> lines)
+        {
+            if (LineFeedMode == LineFeedMode.Wrap || LineFeedMode == LineFeedMode.WordWrap)
+            {
+                int roomOnLine = Size.Width - cursorPos;
+                roomOnLine = Math.Max(roomOnLine, 0);
+                for (int i = 0; i < lines.Count; i++)
+                {
+                    string remaining = lines[i];
+                    if (remaining.Length > roomOnLine)
+                    {
+                        if (LineFeedMode == LineFeedMode.WordWrap)
+                        {
+                            // In intelligent mode, try finding spaces backwards
+                            // Note: this indexes the element 1 char after the last to be printed in the first iteration, since there might be a space just there
+                            int tempRoomOnLine = roomOnLine;
+                            while (remaining[tempRoomOnLine] != ' ' && tempRoomOnLine > 0)
+                            {
+                                tempRoomOnLine--;
+                            }
+
+                            // If we didn't find a space, break after the maximum number of chars
+                            if (tempRoomOnLine > 0)
+                            {
+                                roomOnLine = tempRoomOnLine;
+                            }
+                        }
+
+                        // This line does not fit on the remaining room of this row
+                        string firstPart = remaining.Substring(0, roomOnLine);
+
+                        lines[i] = firstPart;
+                        // Insert the remaining part as new entry to the list, continue iteration there
+                        string remainingPart = remaining.Substring(roomOnLine).TrimStart(' ');
+                        lines.Insert(i + 1, remainingPart);
+
+                    }
+                    roomOnLine = Size.Width; // From now on, we have the full width available
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes a newline to the display (wrapping to the next line)
+        /// </summary>
+        public void WriteLine()
+        {
+            NewLine();
+        }
+
+        /// <summary>
+        /// Writes the given text to the current position, then wraps to the next line
+        /// </summary>
+        /// <param name="text">Text to draw.</param>
+        public void WriteLine(string text)
+        {
+            Write(text);
+            NewLine();
+        }
+
+        /// <summary>
+        /// Blinks the display text (and the backlight, if available).
+        /// Can be used to get user attention. 
+        /// Operation is synchronous.
+        /// </summary>
+        /// <param name="times">Number of times to blink. The blink rate is 1 Hz</param>
+        public void BlinkDisplay(int times)
```

Comment:
this one makes sense to make async version directly and for sync version simply wait on the async